boards/arm/adafruit_trinket_m0/dts.fixup

```diff
@@ -0,0 +1,54 @@
+#define CONFIG_GPIO_SAM0_PORTA_LABEL ATMEL_SAM0_GPIO_41004400_LABEL
```

Comment:
Lets refactor the atmel sam0 dts.fixup into arch/arm/soc/

Reply:
Done.  Note that the samd/ version wasn't getting picked up so I moved it to samd21/